content/blog/2021/10/new-architecture-to-ease-kubewarden-administrators-lives.md

```diff
@@ -0,0 +1,257 @@
+
+For a comprehensive list, see the [release notes](TODO) of our latest Helm
+charts.
+
+## How does it look like?
+
+In previous versions, the Kubewarden Controller instantiated a single Deployment
+of policy-server. That policy-server was configured via a ConfigMap, which
+contained the Deployment options (image, replicas, …), and a list of policies to
+load compiled from the instantiated ClusterAdmissionPolicies, with information
+on where to pull them from, their configuration options and so on.
+
+With the addition of the new **PolicyServer** Custom Resource, administrators
+have a better UX (just define the Custom Resources), can define more than 1
+policy-server Deployment, and get to select what ClusterAdmissionPolicies target
+which PolicyServers. Let's see a diagram of the new architecture:
+
+{{< figure class="center" src="/images/how-it-works-kubewarden.svg" width="90%" alt="new architecture diagram">}}
+
+On the diagram, notice the 2 separate PolicyServer Deployments in cyan and mauve
+(right), created as specified in the 2 PolicyServer resources (left).
+
+Each policy server loads different policies, specified in the
+ClusterAdmissionPolicies. The new PolicyServer Custom Resource is cluster-wide,
+which means that is identifiable by its unique `name`. Here is an example of a
+PolicyServer named `tenant-a`:
+
+```yaml
+---
+apiVersion: policies.kubewarden.io/v1alpha2
+kind: PolicyServer
+metadata:
+  name: tenant-a
+spec:
+  image: ghcr.io/kubewarden/policy-server:latest
+  replicas: 1
+```
+
+A ClusterAdmissionPolicy targeting that PolicyServer needs to set
+`spec.policyServer` to `tenant-a`, as such:
+
+```yaml
+---
+apiVersion: policies.kubewarden.io/v1alpha2
+kind: ClusterAdmissionPolicy
+metadata:
+  name: psp-capabilities
+spec:
+  policyServer: tenant-a
+  module: registry://ghcr.io/kubewarden/policies/psp-capabilities:v0.1.3
+  rules:
+    - apiGroups: [""]
+      apiVersions: ["v1"]
+      resources: ["pods"]
+      operations:
+        - CREATE
+        - UPDATE
+  mutating: true
+  settings:
+    allowed_capabilities:
+      - CHOWN
+    required_drop_capabilities:
+      - NET_ADMIN
+```
+
+For a more in depth dive, have a look at our [architecture
+docs](https://docs.kubewarden.io/architecture.html), and the [Kubewarden CRDs documentation](https://github.com/kubewarden/kubewarden-controller/blob/main/docs/crds/README.asciidoc).
+
+## Ok, but what does this mean for administrators?
+
+Moving from 1 PolicyServer to several increases resiliency and segregation. Now,
+we can take admission reviews from policies and separate them into several
+PolicyServers. While the old architecture was already HA, a noisy
+tenant/namespace or a frequently used policy could bring to a crawl the only
+policy server and break havoc in the Cluster. With the new possibility of
+several PolicyServers, a Kubernetes Operator can isolate policy evaluations per
+tenant/namespace, and run mission critical policies separately, making the whole
```

Comment:
In general, they can isolate as they want literally, not specifically tenant/namespace, although it will be surely a very common pattern. I think we should somehow point out that how the given ClusterAdmissionPolicy targets specific PolicyServer (and namespaced AdmissionPolicy in the future), is completely up to the user, and how they want to split/organize is completely up to the cluster administrators.

Reply:
True. I reworded and added a couple of paragraphs making that distinction in https://github.com/kubewarden/kubewarden.io/pull/33/commits/fe3faa37d1105e3cb847c9d79b24911697ec20be.